Covid vaccine expansion

cohnjesse opened the pull request:
## Description
Now that we have a basic design for the covid vaccine eligibility expansion work we need to get the basics of the form set up. This PR creates a new, expanded form with all of the pages as well as some scaffolding of some custom functionality we will need. All of the pages are still being refined in terms of design so there will be changes.

vets-json-schema PR: department-of-veterans-affairs/vets-json-schema#570

## Testing done
Unit and e2e tests will be included in a subsiquent PR

## Acceptance criteria
- [x] The covid eligibility expansion form has been started
- [x] A PR has been submitted for review

Commit:
Added form

Commit:
Added all pages

stephen-barrs-va committed:
intro page updates for 3 buttons

Commit:
removed extra file

Commit:
Merge branch 'covid-vaccine-expansion' of github.com:department-of-veterans-affairs/vets-website into covid-vaccine-expansion

Commit:
filled out address page

Commit:
Added not eligible

stephen-barrs-va committed:
radio buttons on intro page

Commit:
messing with location stuff

Commit:
Merge branch 'covid-vaccine-expansion' of github.com:department-of-veterans-affairs/vets-website into covid-vaccine-expansion

Commit:
Updated contact and address pages

stephen-barrs-va committed:
statement of truth with static values

stephen-barrs-va committed:
dynamic statement of truth

Commit:
Added  static location component

Commit:
fleshed out static location finder

stephen-barrs-va committed:
conditional required for the statement of truth based on selected user type

stephen-barrs-va committed:
attestation Additional Info adds

Commit:
Added military history and redux

Commit:
Merge branch 'covid-vaccine-expansion' of github.com:department-of-veterans-affairs/vets-website into covid-vaccine-expansion

# Conflicts:
#	src/applications/covid-vaccine/config/pages.js

stephen-barrs-va committed:
verbiage for none of above

Commit:
fixed import

Commit:
Fixed linting issues

stephen-barrs-va committed:
none of the above in attestation

stephen-barrs-va committed:
resolve merge conflict

bkjohnson committed:
Use custom widget for VA location

bkjohnson committed:
Avoid changing forms library files

bkjohnson committed:
Connect component to redux store

This is so that we can get the zipcode from the form data and include it
in the api call

Commit:
Updated to use local API call

stephen-barrs-va committed:
merge dynamic checkbox work

stephen-barrs-va committed:
refactor radios and button to use storybook components

stephen-barrs-va committed:
style radio group to remove large space

stephen-barrs-va committed:
update form page routes

stephen-barrs-va committed:
update branch of service and discharge types

Commit:
Fleshed out content in widget

stephen-barrs-va committed:
adding presubmit info checkbox to review page

bkjohnson committed:
Add monthYear range to military history

bkjohnson committed:
Don't make mobile number required

stephen-barrs-va committed:
privacy statement modal on intro page

stephen-barrs-va committed:
merging latest

bkjohnson committed:
Remove spouse's suffix

bkjohnson committed:
Merge branch 'covid-vaccine-expansion' of github.com:department-of-veterans-affairs/vets-website into covid-vaccine-expansion

Commit:
Changed app folder name

bkjohnson committed:
Refactor required fields

bkjohnson committed:
Make street address required

Commit:
Added error states to location page

Commit:
changed error

Commit:
Merge branch 'bj-covid-vaccine-expansion' into covid-vaccine-expansion

# Conflicts:
#	src/applications/coronavirus-vaccination-expansion/config/military-history/helper.js
#	src/applications/coronavirus-vaccination-expansion/containers/privacyDataHelper.js

stephen-barrs-va committed:
privacy modal update and remove disable from button

stephen-barrs-va committed:
merging

bkjohnson committed:
Add error message on intro page

bkjohnson committed:
Merge branch 'covid-vaccine-expansion' of github.com:department-of-veterans-affairs/vets-website into covid-vaccine-expansion

bkjohnson committed:
Fix a routing problem & update manifest

stephen-barrs-va committed:
add conditional page for spouse and caregivers to enter Veterans info

Commit:
Added birth question

Commit:
Merge branch 'covid-vaccine-expansion' of github.com:department-of-veterans-affairs/vets-website into covid-vaccine-expansion

bkjohnson committed:
Update mobile phone field description

bkjohnson committed:
Try to make the AdditionalInfo more accessible

Commit:
Updated package.json and imported schemas

bkjohnson committed:
Fix typo on addressInformation page

Commit:
Merge branch 'covid-vaccine-expansion' of github.com:department-of-veterans-affairs/vets-website into covid-vaccine-expansion

# Conflicts:
#	src/applications/coronavirus-vaccination-expansion/config/address/addressInformation.js

Commit:
Merge branch 'covid-vaccine-expansion' of github.com:department-of-veterans-affairs/vets-website into covid-vaccine-expansion

bkjohnson committed:
Merge branch 'master' into covid-vaccine-expansion

bkjohnson committed:
Fix broken merge

bkjohnson committed:
Remove mock-api file

bkjohnson committed:
Remove old schema file

I think this has been replaced by stuff in vets-json-schema

bkjohnson commented on src/applications/coronavirus-vaccination-expansion/config/military-history/helper.js:
```diff
@@ -0,0 +1,32 @@
+  undesirable: 'Undesirable',
+};
+
+const lastServiceBranchLabels = {
+  'air force': 'Air Force',
+  army: 'Army',
+  'coast guard': 'Coast Guard',
+  'marine corps': 'Marine Corps',
+  'merchant seaman': 'Merchant Seaman',
+  navy: 'Navy',
+  noaa: 'Noaa',
+  usphs: 'USPHS',
+  'f.commonwealth': 'Filipino Commonwealth Army',
+  'f.guerilla': 'Filipino Guerilla Forces',
+  'f.scouts new': 'Filipino New Scout',
+  'f.scouts old': 'Filipino Old Scout',
+  other: 'Other',
+};
```
Can we make use of the platform const here:

https://github.com/department-of-veterans-affairs/vets-website/blob/0677518c2bd7a46e665cd1a54e3c32cb15a63a4e/src/platform/forms/address/data/labels.js#L6-L12

This seems even better:

https://github.com/department-of-veterans-affairs/vets-website/blob/0677518c2bd7a46e665cd1a54e3c32cb15a63a4e/src/platform/static-data/options-for-select.js#L9-L23

bkjohnson commented on src/applications/coronavirus-vaccination-expansion/config/military-history/helper.js:
```diff
@@ -0,0 +1,32 @@
+const dischargeTypeLabels = {
+  honorable: 'Honorable',
+  general: 'General',
+  other: 'Other Than Honorable',
+  'bad-conduct': 'Bad Conduct',
+  dishonorable: 'Dishonorable',
+  undesirable: 'Undesirable',
+};
```
Importing these seems like a good idea rather than redefining them:

https://github.com/department-of-veterans-affairs/vets-website/blob/0677518c2bd7a46e665cd1a54e3c32cb15a63a4e/src/platform/static-data/options-for-select.js#L25-L32

bkjohnson committed:
Remove unused helper file

bkjohnson committed:
Remove unused consts

These seem to have been replaced by vets-json-schema imports in the
addressSchema file

bkjohnson committed:
Remove location-finder

Nothing is using this right now, but we'll need to add that markup back
in somewhere to match the design

bkjohnson committed:
Refactor the body copy for the VA locations

stephen-barrs-va committed:
add schema name to Contstants file

bkjohnson commented on src/applications/coronavirus-vaccination-expansion/config/personal-information/personal-information.js:
```diff
@@ -0,0 +1,79 @@
+
+export const uiSchema = {
+  personalInformation: {
+    firstName: {
+      'ui:title': 'First name',
+      'ui:errorMessages': {
+        required: 'Please enter your first name.',
+      },
+    },
+    middleName: {
+      'ui:title': 'Middle name',
+    },
+    lastName: {
+      'ui:title': 'Last name',
+      'ui:errorMessages': {
+        required: 'Please enter your last name.',
+      },
+    },
```
We can probable reuse the `fullName` definition.

https://github.com/department-of-veterans-affairs/vets-website/blob/covid-vaccine-expansion/src/platform/forms/definitions/fullName.js

batemapf committed:
add our dir to codeowners

Commit:
Added root url

Commit:
Merge branch 'covid-vaccine-expansion' of github.com:department-of-veterans-affairs/vets-website into covid-vaccine-expansion

stephen-barrs-va committed:
add empty SIP for tests

batemapf committed:
content and ui tweaks

stephen-barrs-va committed:
use discharge types and branches from common

stephen-barrs-va committed:
add uiSchema field for sms ack

stephen-barrs-va committed:
prevent future DOB

stephen-barrs-va committed:
update vets json schema ref

stephenBarrs commented:
vets-json-schema PR: https://github.com/department-of-veterans-affairs/vets-json-schema/pull/570

theodur commented on src/applications/registry.json:
```diff
@@ -1002,6 +1002,15 @@
     }
   },
+  {
+    "appName": "covid-vaccine-expansion",
```
The `appName` is used as the title of the HTML page that is generated. Do we want to update the formatting to read `Covid Vaccine Expansion`?

theodur commented on src/applications/coronavirus-vaccination-expansion/manifest.json:
```diff
@@ -0,0 +1,6 @@
+{
+  "appName": "covid-vaccine-expansion",
```
The same case here, if we update the `appName` in the `registry.json` file we'd probably want to do that here as well.

bkjohnson commented on src/platform/forms/constants.js:
```diff
@@ -25,6 +25,7 @@ export const VA_FORM_IDS = Object.freeze({
   FORM_COVID_VACCINE_TRIAL: 'COVID-VACCINE-TRIAL',
   FORM_21_22: '21-22',
   FORM_5655: '5655',
+  FORM_COVID_VACCINATION_EXPANSION: 'COVID-VACCINATION-EXPANSION',
```
This is the only thing I'm a codeowner of - no objections.

batemapf committed:
update schema ref

Commit:
Bumped vets-json-schema version
